File: intralogin/app/javascript/channels/user_status_channel.js
import consumer from "./consumer"
import $ from "jquery"
import Backbone from "backbone"

consumer.subscriptions.create("UserStatusChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("Welcome User!")
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
    console.log("Bye User!")
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  }
});

File: intralogin/app/channels/user_status_channel.rb
class UserStatusChannel < ApplicationCable::Channel
  def subscribed
    user_status = User.find_by(id: current_user[:id])
    user_status.update!(status: 1)
    data = {:id => current_user[:id], :status => 1}
    stream_from "user_status_channel"
    ActionCable.server.broadcast "user_status_channel", data
  end

  def unsubscribed
    user_status = User.find_by(id: current_user[:id])
    user_status.update!(status: 0)
    data = {:id => current_user[:id], :status => 0}
    ActionCable.server.broadcast "user_status_channel", data
  end
end
